src/Microsoft.Sbom.Api/Workflows/Helpers/FileArrayGenerator.cs

```diff
@@ -47,24 +48,29 @@ public FileArrayGenerator(
     /// <param name="jsonSerializer">The serializer used to write the SBOM.</param>
     /// <param name="headerName">The header key for the file array object.</param>
     /// <returns></returns>
-    public async Task<GenerationResult> GenerateAsync()
+    public async Task<GenerationResult> GenerateAsync(IList<ManifestInfo> manifestInfosFromConfig, HashSet<string> elementsSpdxIdList)
     {
         using (recorder.TraceEvent(Events.FilesGeneration))
         {
             var totalErrors = new List<FileValidationResult>();
 
-            var sourcesProviders = this.sourcesProviders
-                .Where(s => s.IsSupported(ProviderType.Files));
-
             // Write the start of the array, if supported.
             IList<ISbomConfig> filesArraySupportingSboms = new List<ISbomConfig>();
-            var serializationStrategy = JsonSerializationStrategyFactory.GetStrategy(SpdxManifestVersion);
-            var jsonArrayStarted = serializationStrategy.AddToFilesSupportingConfig(filesArraySupportingSboms, this.SbomConfig);
+            var jsonArrayStartedForConfig = new Dictionary<ISbomConfig, bool>();
+            var filesSourcesProviders = this.sourcesProviders
+               .Where(s => s.IsSupported(ProviderType.Files));
 
-            this.logger.Verbose("Started writing files array for {configFile}.", this.SbomConfig.ManifestJsonFilePath);
+            foreach (var manifestInfo in manifestInfosFromConfig)
+            {
+                var config = sbomConfigs.Get(manifestInfo);
```

Comment:
I believe that `manifestInfosFromConfig` is unfiltered. Should we use a `TryGet` to protect against unregistered configs?